mobify/platform-scaffold
# [WEB-1395]: Prefetch Template Chunks

Previously, our code-split chunks were fetched AND loaded.

We now prefetch them instead, so that they are not loaded until they are needed.

 **JIRA**: https://mobify.atlassian.net/browse/WEB-1395
 **Linked PRs**: N/A

## Changes
- Prefetch instead of preload code-split chunks

## Todos
- [x] Test changes on iOS and Android Devices

## How to test-drive this PR
- Run `npm run dev`
- Preview [Merlin's Potions](https://preview.mobify.com/?url=https%3A%2F%2Fwww.merlinspotions.com%2F&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1)
- Before navigating... to ensure that the coded chunks were loaded correctly use the chrome dev tools:
    1. Check the network tab to see the requests for template files: 
        - `cart.js`
        - `checkout-confirmation.js`
        - `checkout-payment.js`
        - `checkout-shipping.js`
        - `login.js`
        - `product-details.js`
        - `product-list.js`
    2. Check that they are all cached `Application` > `Cache` > `progressive-web-scaffold-<version>-bundle`
- Cut the network connection.
- Navigate to a PLP, and it should still fail gracefully.

## Results
Webpage tests result in a more consistent measurement of TTI ~ 9.9 - 10.5.
https://www.webpagetest.org/result/170619_WM_1PFE/

The fetched code chunks no longer execute when fetched from the network.
Turns out something else is blocking the main thread so we are still in the 10s TTI range because of it.
